test/jdk/java/awt/TrayIcon/ShowAfterDisposeTest/ShowAfterDisposeTest.java

```diff
@@ -22,67 +22,72 @@
  */
 
 /*
-  test
-  @bug 6384984 8004032
-  @summary TrayIcon try to dispay a tooltip when is not visible
-  @author Dmitry.Cherepanov@sun.com area=awt.tray
-  @run applet/manual=yesno ShowAfterDisposeTest.html
+ * @test
+ * @bug 6384984 8004032
+ * @library ../../regtesthelpers
+ * @build PassFailJFrame
+ * @summary TrayIcon try to dispay a tooltip when is not visible
+ * @run main/manual ShowAfterDisposeTest
 */
 
-import java.applet.*;
-
-import java.awt.*;
-import java.awt.event.*;
-import java.awt.image.*;
-
-public class ShowAfterDisposeTest extends Applet
-{
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.SystemTray;
+import java.awt.TrayIcon;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.image.BufferedImage;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+
+public class ShowAfterDisposeTest {
+    JFrame frame;
     boolean traySupported;
 
-    public void init()
-    {
-        this.setLayout (new BorderLayout ());
+    public static void main(String[] args) throws Exception {
+        ShowAfterDisposeTest test = new ShowAfterDisposeTest();
+        test.startTest();
+    }
 
-        String[] instructions;
+    public void startTest() throws Exception {
         traySupported = SystemTray.isSupported();
-        if (traySupported)
-        {
-            String[] s =
-            {
-                "1) When the test starts an icon is added to the SystemTray area.",
-                "2a) If you use Apple OS X,",
-                "    right click on this icon (it's important to click before the tooltip is shown).",
-                "    The icon should disappear.",
-                "2b) If you use other os (Windows, Linux, Solaris),",
-                "    double click on this icon (it's important to click before the tooltip is shown).",
-                "    The icon should disappear.",
-                "3) If the bug is reproducible then the test will fail without assistance.",
-                "4) Just press the 'pass' button."
-            };
-            instructions = s;
-        }
-        else
-        {
-            String[] s =
-            {
-              "The test cannot be run because SystemTray is not supported.",
-              "Simply press PASS button."
-            };
-            instructions = s;
+        if (!traySupported) {
+            System.out.println("The test cannot be run because SystemTray is not supported.");
+            return;
         }
```

Comment:
I'd rather move this entire check into the `main` method and throw `jtreg.SkippedException` instead.